pkg/provisioner/ironic/ironic.go

```diff
@@ -321,6 +321,53 @@ func (p *ironicProvisioner) getNode() (*nodes.Node, error) {
 	}
 }
 
+// Verifies that node has port assigned by Ironic.
+func (p *ironicProvisioner) nodeHasAssignedPort(ironicNode *nodes.Node) (bool, error) {
+	opts := ports.ListOpts{
+		Fields:   []string{"node_uuid"},
+		NodeUUID: ironicNode.UUID,
+	}
```

Comment:
It feels like we're copying and pasting all of the logic of listAllPorts() here for the sake of not adding one extra filtering option parameter.